nixos/modules/services/hardware/keyd.nix

```diff
@@ -82,21 +96,23 @@ in
 
   config = mkIf cfg.enable {
     # Creates separate files in the `/etc/keyd/` directory for each key in the dictionary
-    environment.etc = mapAttrs'
-      (name: options:
-        nameValuePair "keyd/${name}.conf" {
-          source = pkgs.runCommand "${name}.conf"
-            {
-              ids = ''
+    environment.etc =
+      mapAttrs'
+        (name: options:
+          nameValuePair "keyd/${name}.conf" {
+            text =
+              let
+                settings = settingsFormat.generate "keyd-${name}.conf" options.settings;
+              in
+              ''
                 [ids]
                 ${concatStringsSep "\n" options.ids}
+
+                ${builtins.readFile settings}
```

Comment:
This is an [IFD](https://nixos.wiki/wiki/Import_From_Derivation).

I am not sure if it is fine to use IFD in NixOS modules.  I find an [example](https://github.com/NixOS/nixpkgs/pull/189587) which removes its usage.